stream_alert/alert_processor/outputs/output_base.py

```diff
@@ -194,46 +172,75 @@ def _load_creds(self, descriptor):
-        defaults = self._get_default_properties()
+        defaults = self._defaults
         if defaults:
             creds_dict.update(defaults)
 
         return creds_dict
 
-    def _get_creds_from_s3(self, cred_location, descriptor):
+    @staticmethod
+    def get_s3_secrets_bucket(prefix):
+        return '{}.streamalert.secrets'.format(prefix)
+
+    @staticmethod
+    def get_local_credentials_temp_dir():
+        """Get the local tmp directory for caching the encrypted service credentials.
+           Will automatically create a new directory
+
+        Returns:
+            str: local path for stream_alert_secrets tmp directory
+        """
+        temp_dir = os.path.join(tempfile.gettempdir(), "stream_alert_secrets")
+
+        # Check if this item exists as a file, and remove it if it does
+        if os.path.isfile(temp_dir):
+            os.remove(temp_dir)
+
+        # Create the folder on disk to store the credentials temporarily
+        if not os.path.exists(temp_dir):
+            os.makedirs(temp_dir)
+
+        return temp_dir
+
+    def load_encrypted_credentials_from_s3(self, cred_location, descriptor):
         """Pull the encrypted credential blob for this service and destination from s3
+           and save it to a local file.
 
         Args:
             cred_location (str): The tmp path on disk to to store the encrypted blob
             descriptor (str): Service destination (ie: slack channel, pd integration)
 
         Returns:
-            bool: True if download of creds from s3 was a success
+            bool: True if credentials are downloaded from S3 successfully.
         """
         try:
             if not os.path.exists(os.path.dirname(cred_location)):
                 os.makedirs(os.path.dirname(cred_location))
 
-            client = boto3.client('s3', region_name=self.region)
+            client = boto3.client('s3', region_name=self._region)
             with open(cred_location, 'wb') as cred_output:
-                client.download_fileobj(self.secrets_bucket,
-                                        self.output_cred_name(descriptor),
-                                        cred_output)
+                client.download_fileobj(
+                    self._secrets_bucket,
+                    self.get_formatted_output_credentials_name(self._service_name, descriptor),
+                    cred_output
+                )
 
             return True
         except ClientError as err:
             LOGGER.exception('credentials for \'%s\' could not be downloaded '
-                             'from S3: %s', self.output_cred_name(descriptor),
+                             'from S3: %s',
+                             self.get_formatted_output_credentials_name(self._service_name,
+                                                                        descriptor),
                              err.response)
 
-    def _kms_decrypt(self, data):
+    def kms_decrypt(self, data):
```

Comment:
I'm sure there is a reason for this but I can't find it.. why the change to a public method from protected?

Reply:
This method gets nuked in #878